httpx/dispatch/connection_pool.py

```diff
@@ -124,7 +125,11 @@ def num_connections(self) -> int:
             self.active_connections.remove(connection)
             self.max_connections.release()
             raise exc
-
+        except NewConnectionRequired:
+            # raised when stream ids expire
```

Comment:
We shouldn't let this HTTP/2 detail surface out here — it's probably OK to remove this comment.

```suggestion
```